robot_skills/src/robot_skills/functionalities/arm/guarded_motion.py

```diff
@@ -11,55 +11,77 @@ class GuardedMotionFunc(RobotFunc):
     def __init__(self):
         super(GuardedMotionFunc, self).__init__("guarded_motion",
                                                 Arm,
-                                                {"move_down_until_force_sensor_edge_up": move_down_until_force_sensor_edge_up,
-                                                 "_create_lower_force_sensing_goal": _create_lower_force_sensing_goal,
-                                                 "_create_retract_force_sensing_goal": _create_retract_force_sensing_goal})
+                                                {
+                                                    "move_down_until_force_sensor_edge_up": move_down_until_force_sensor_edge_up,
+                                                    "_create_lower_force_sensing_goal": _create_lower_force_sensing_goal,
+                                                    "_create_retract_force_sensing_goal": _create_retract_force_sensing_goal
+                                                })
 
     def check_requirements(self, arm):
-        "Check that the arm has at least one force sensor"
+        """
+        Check that the arm has at least one force sensor
+        """
         for part in arm.parts:
             if isinstance(arm.parts[part], ForceSensor):
                 return True
         return False
 
 
-def move_down_until_force_sensor_edge_up(self, force_sensor=None, timeout=10, retract_distance=0.01):
+def move_down_until_force_sensor_edge_up(self, force_sensor=None, timeout=10, retract_distance=0.01,
+                                         distance_move_down=None):
     """
-    Move down the arm (hero specific, only joint arm_lift_joint) until the force sensor detects an edge up
+    Move down the arm (hero specific, only joint arm_lift_joint) until one of 3 things:
+        - Force sensor detects an edge up
+        - Timeout
+        - Force sensor has moved down the distance specified
 
     A 'force_sensor.TimeOutException' will be raised if no edge up is detected within timeout
 
+    :param force_sensor: ForceSensor of the arm
     :param timeout: Max duration for edge up detection
     :param retract_distance: How much to retract if we have reached a surface
+    :param distance_move_down: How much to move down until we reach a surface
     """
-    if not force_sensor:
+    if force_sensor is None:
         force_sensor = self.parts["force_sensor"]
 
     # Fill with required joint names (desired in hardware / gazebo impl)
-    goal = self._create_lower_force_sensing_goal(timeout)
+    goal = self._create_lower_force_sensing_goal(distance_move_down, timeout)
     self._ac_joint_traj.send_goal(goal)
 
     force_sensor.wait_for_edge_up(timeout)
```

Comment:
The behaviour we currently create will wait for the timeout if its end goal is reached. So it might be waiting at the lower position without continuing.